### PR TITLE
std/http/server::serve aligned to std/http/server::serveTLS

### PR DESCRIPTION
**Pull request by kitsonk**
Previously, `std/http::serve()` previously create a new interface which it didn't export that was the same as `Deno.ListenOptions`.  It is better that it just use the built in interface.

### CODE REVIEW

File: std/http/server.ts [hunk 1]
```diff
@@ -483,7 +478,7 @@ interface ServerConfig {
  *       req.respond({ body });
  *     }
  */
-export function serve(addr: string | ServerConfig): Server {
+export function serve(addr: string | Deno.ListenOptions): Server {
```
Comment (bartlomieju) [inline std/http/server.ts]: :+1: can you apply the same rule to `serveTLS` and  `listenAndServeTLS`?

Comment (kitsonk) [reply std/http/server.ts]: Are you sure?  From the code:

```ts
export type HTTPSOptions = Omit<Deno.ListenTLSOptions, "transport">;
```

It is exported, and it is different that `Deno.ListenTLSOptions` intentionally.

Comment (bartlomieju) [reply std/http/server.ts]: You're right, I see it skips "protocol" field - then I guess it makes sense to apply the same for `serve`.

Comment (kitsonk) [reply std/http/server.ts]: yeah, let me align them...

**Pull request merged**